src/backend/cdb/cdbpartition.c

```diff
@@ -6771,6 +6771,23 @@ atpxPartAddList(Relation rel,
 	if (pelem->storeAttr)
 		ct->options = (List *) ((AlterPartitionCmd *) pelem->storeAttr)->arg1;
 
+	if (gp_add_partition_inherits_table_setting)
+	{
+		ListCell *lc;
+		List *opts = reloptions_list(RelationGetRelid(rel));
+
+		foreach(lc, ct->options)
+		{
+			DefElem *de = lfirst(lc);
+			DefElem *def = reloptions_get_opt(opts, de->defname);
+
+			if (def != NULL)
+				*def = *de;
+		}
+
+		ct->options = build_ao_rel_storage_opts(opts, rel);
```

Comment:
Why do we need use `build_ao_rel_storage_opts`? It looks like everything works if you just copy `opts`. And we won't get any extra parameters in the output `\d`

Reply:
> Why do we need use `build_ao_rel_storage_opts`?

Yes

> It looks like everything works if you just copy `opts`.

No

Without build_ao_rel_storage_opts not explicit pointed options are taken from default instead from table.

Reply:
> we won't get any extra parameters in the output `\d`

unfortunately we can not avoid this

Reply:
> Without build_ao_rel_storage_opts not explicit pointed options are taken from default instead from table.

I checked and not explicitly pointed options are taken from the parent table. if just copy opts

Reply:
> I checked and not explicitly pointed options are taken from the parent table. if just copy opts

Really?!
```diff
diff --git a/src/backend/cdb/cdbpartition.c b/src/backend/cdb/cdbpartition.c
index 104ddbceed..efc5a08c59 100644
--- a/src/backend/cdb/cdbpartition.c
+++ b/src/backend/cdb/cdbpartition.c
@@ -6785,7 +6785,7 @@ atpxPartAddList(Relation rel,
 				*def = *de;
 		}
 
-		ct->options = build_ao_rel_storage_opts(opts, rel);
+		ct->options = opts;
 	}
 
 	ct->tableElts = list_concat(ct->tableElts, list_copy(colencs));
diff --git a/src/test/regress/expected/alter_table_ao.out b/src/test/regress/expected/alter_table_ao.out
index e1c9fc6317..1a4f546f0b 100644
--- a/src/test/regress/expected/alter_table_ao.out
+++ b/src/test/regress/expected/alter_table_ao.out
@@ -832,7 +832,7 @@ Check constraints:
     "ao_alter_add_part_1_prt_40_check" CHECK (b >= 30 AND b < 40)
 Inherits: ao_alter_add_part
 Distributed by: (a)
-Options: appendonly=true, orientation=row, compresstype=zstd, compresslevel=4, blocksize=65536, checksum=true
+Options: appendonly=true, orientation=row, compresstype=zstd, compresslevel=4, blocksize=65536
 
         Append-Only Table "public.ao_alter_add_part_1_prt_50"
  Column |  Type   | Modifiers | Storage | Stats target | Description 
@@ -847,7 +847,7 @@ Check constraints:
     "ao_alter_add_part_1_prt_50_check" CHECK (b >= 40 AND b < 50)
 Inherits: ao_alter_add_part
 Distributed by: (a)
-Options: appendonly=true, orientation=row, compresstype=zstd, compresslevel=4, blocksize=65536, checksum=true
+Options: appendonly=true, orientation=row, compresstype=zstd, compresslevel=4, blocksize=65536
 
         Append-Only Table "public.ao_alter_add_part_1_prt_60"
  Column |  Type   | Modifiers | Storage | Stats target | Description 
@@ -953,22 +953,22 @@ Options: appendonly=true, orientation=row, compresstype=zstd, compresslevel=3, b
 --------+---------+-----------+---------+--------------+-------------
  a      | integer |           | plain   |              | 
  b      | integer |           | plain   |              | 
-Compression Type: none
-Compression Level: 0
-Block Size: 32768
+Compression Type: zlib
+Compression Level: 2
+Block Size: 8192
 Checksum: t
 Check constraints:
     "ao_alter_add_part_no_compress_1_prt_40_check" CHECK (b >= 30 AND b < 40)
 Inherits: ao_alter_add_part_no_compress
 Distributed by: (a)
-Options: appendonly=true, orientation=row, blocksize=32768, compresslevel=0, checksum=true, compresstype=none
+Options: appendonly=true, orientation=row, blocksize=8192, compresslevel=2
 
   Append-Only Table "public.ao_alter_add_part_no_compress_1_prt_50"
  Column |  Type   | Modifiers | Storage | Stats target | Description 
 --------+---------+-----------+---------+--------------+-------------
  a      | integer |           | plain   |              | 
  b      | integer |           | plain   |              | 
-Compression Type: none
+Compression Type: None
 Compression Level: 0
 Block Size: 32768
 Checksum: t
@@ -976,7 +976,7 @@ Check constraints:
     "ao_alter_add_part_no_compress_1_prt_50_check" CHECK (b >= 40 AND b < 50)
 Inherits: ao_alter_add_part_no_compress
 Distributed by: (a)
-Options: appendonly=true, orientation=row, blocksize=32768, compresslevel=0, checksum=true, compresstype=none
+Options: appendonly=true, orientation=row
 
   Append-Only Table "public.ao_alter_add_part_no_compress_1_prt_60"
  Column |  Type   | Modifiers | Storage | Stats target | Description 
diff --git a/src/test/regress/expected/alter_table_aocs.out b/src/test/regress/expected/alter_table_aocs.out
index 70de68f504..486caac628 100644
--- a/src/test/regress/expected/alter_table_aocs.out
+++ b/src/test/regress/expected/alter_table_aocs.out
@@ -926,7 +926,7 @@ Check constraints:
     "aocs_alter_add_part_1_prt_40_check" CHECK (b >= 30 AND b < 40)
 Inherits: aocs_alter_add_part
 Distributed by: (a)
-Options: appendonly=true, orientation=column, compresstype=rle_type, compresslevel=4, blocksize=65536, checksum=true
+Options: appendonly=true, orientation=column, compresstype=rle_type, compresslevel=4, blocksize=65536
 
                             Append-Only Columnar Table "public.aocs_alter_add_part_1_prt_50"
  Column |  Type   | Modifiers | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
@@ -938,7 +938,7 @@ Check constraints:
     "aocs_alter_add_part_1_prt_50_check" CHECK (b >= 40 AND b < 50)
 Inherits: aocs_alter_add_part
 Distributed by: (a)
-Options: appendonly=true, orientation=column, compresstype=rle_type, compresslevel=4, blocksize=65536, checksum=true
+Options: appendonly=true, orientation=column, compresstype=rle_type, compresslevel=4, blocksize=65536
 
                             Append-Only Columnar Table "public.aocs_alter_add_part_1_prt_60"
  Column |  Type   | Modifiers | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
@@ -1021,14 +1021,14 @@ Options: appendonly=true, orientation=column, compresstype=rle_type, compresslev
                       Append-Only Columnar Table "public.aocs_alter_add_part_no_compress_1_prt_40"
  Column |  Type   | Modifiers | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
 --------+---------+-----------+---------+--------------+------------------+-------------------+------------+-------------
- a      | integer |           | plain   |              | none             | 0                 | 32768      | 
- b      | integer |           | plain   |              | none             | 0                 | 32768      | 
+ a      | integer |           | plain   |              | zlib             | 2                 | 8192       | 
+ b      | integer |           | plain   |              | zlib             | 2                 | 8192       | 
 Checksum: t
 Check constraints:
     "aocs_alter_add_part_no_compress_1_prt_40_check" CHECK (b >= 30 AND b < 40)
 Inherits: aocs_alter_add_part_no_compress
 Distributed by: (a)
-Options: appendonly=true, orientation=column, blocksize=32768, compresslevel=0, checksum=true, compresstype=none
+Options: appendonly=true, orientation=column, blocksize=8192, compresslevel=2
 
                       Append-Only Columnar Table "public.aocs_alter_add_part_no_compress_1_prt_50"
  Column |  Type   | Modifiers | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
@@ -1040,7 +1040,7 @@ Check constraints:
     "aocs_alter_add_part_no_compress_1_prt_50_check" CHECK (b >= 40 AND b < 50)
 Inherits: aocs_alter_add_part_no_compress
 Distributed by: (a)
-Options: appendonly=true, orientation=column, blocksize=32768, compresslevel=0, checksum=true, compresstype=none
+Options: appendonly=true, orientation=column
 
                       Append-Only Columnar Table "public.aocs_alter_add_part_no_compress_1_prt_60"
  Column |  Type   | Modifiers | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
```